frontend/model/contracts/voting/proposals.js

```diff
@@ -138,8 +108,15 @@ const proposals = {
         [RULE_DISAGREEMENT]: { threshold: 1 }
       }
     },
-    [VOTE_FOR]: function (state, { proposalHash, passPayload }) {
-      console.error('unimplemented!')
+    [VOTE_FOR]: function (state, data) {
+      const proposal = state.proposals[data.proposalHash]
+      proposal.status = STATUS_PASSED
+      const { setting, proposedValue } = proposal.data.proposalData
+
+      sbp('gi.contracts/group/updateSettings/process', state, {
+        meta: proposal.meta,
+        data: { [setting]: proposedValue }
+      })
```

Comment:
@taoeffect I have two questions here:

1. when we do `proposal.status = STATUS_PASSED` we are mutating the state directly. (I copied from INVITE_MEMBER function). Can you remind me why we don't use Vuex mutations to mutate the state?

2. Why do we call sbp `/process` instead of `/create`, as we usually do in other places?

Reply:
Copy/Past @taoeffect's answer from Slack:

> 1. We do use Vuex mutations (as single mutation that does dispatch to sbp). We do it this way because of the SBP integration which gives us greater flexibility
> 2. /create is for creating messages to send. No messages should be sent here. In this case we use /process to DRY.

Okay... I remember now, the mutation is triggered by `'backend/publishLogEntry'` that calls the action `syncContractWithServer`.

2. Why it isn't needed to be sent here?

Reply:
> Okay... I think I remember now, the mutation is triggered by the sbp 'backend/publishLogEntry' that calls the state action syncContractWithServer.

Not exactly. If we're talking about where the Vuex mutation happens, mentioning `'backend/publishLogEntry'` is to go too far back up the chain of events.

`'backend/publishLogEntry'` is what is used to send a GIMessage, which results in a network event. For conceptual purposes, think of that as a totally separate sequence of events. If we want to discuss where the Vuex mutation is, we have to think not in terms of sending messages, but in terms of receiving them.

The conceptual chain of events for receiving messages begins with `'state/enqueueHandleEvent'`, which is called by our websocket when the client receives a message from the server. From there, the Vuex action `handleEvent` gets called, and from there the sequence of events to the Vuex mutation goes like this: `processMutation` => `store.commit(`${contractID}/processMessage`` => `processMessage` => `guardedSBP` => `sbp`

By doing it this way, we decouple our logic from Vuex. We can, if we want to, replace Vuex with any other system. And we can, similarly, turn all of our contract logic into a re-usable library that will work in any project, whether or not they use Vuex, Redux, etc.

> Why it isn't needed to be sent here?

Well think about what is happening. We are receiving and processing message. Why would we need to send any new messages at this point? It makes no sense to send out any other messages.

Reply:
Oh, oops, there is one more thing, and that is you are right to mention `syncContractWithServer`, because that is the other way messages are received in the app.

There are two ways messages are received:

1. Via the websocket (that flow is described in my previous comment above)
2. Via `syncContractWithServer`, which happens in two cases: upon joining a group, and when we sign back into an existing group after being offline for a while.

Both routes will result in calls to `handleEvent` and the chain of events mentioned above.

Reply:
Thanks for the detailed explanation! Definitely I'm saving this comment on my notes!